pkg/cri/resource-manager/cache/container.go

```diff
@@ -253,35 +254,56 @@ func (c *container) GetArgs() []string {
 	return args
 }
 
-func (c *container) GetLabelKeys() []string {
+func (c *container) getLabelKeys(prefix string) []string {
 	keys := make([]string, len(c.Labels))
 
```

Comment:
Sorry if this is getting borderline insane nitpicking, but in principle this code would consider
cri-resource-manager.intel.computer/label as belonging to the cri-resource-manager.intel.com namespace, which I'm guessing is not intentional. This could be easily fixed by adding something like this to the beginning of getLabelKeys():

```
func getLabelKeys(prefix string) []string {
...
    if prefix != "" && !strings.HasSuffix(prefix, "/") {
        prefix += "/"
    }
...
```
Ditto for annotations/getAnnotationKeys().

Reply:
Ha, good that somebody is reading the code :smile: 

`getLabelKeys()` (and `getAnnotationKeys()`) was actually intentionally written this way so that it could be used to filter by actually prefix (instead of namespace), e.g. list all `example.com/cpu*` labels. Makes sense?

Nevertheless, there's a problem with the slashes in this version of the patch: `GetResmgrLabelKeys()` and `GetResmgrAnnotationKeys()` did not append slash to the namespace when composing the prefix. This is now fixed in the latest version of the patchset

Reply:
Ok. Being able to filter based on prefix instead of just namespace does make sense.

Reply:
I just pushed yet another version. After some "proofing time" I started to think something like what you originally suggested was good :smile: At the time being, we want to get keys in a namespace, stripping the namespace part from the results. If we need flexible prefix filtering, that can be easily added later.

Also, the newest version adds some debug printing in the eda policy.